tokeniser.go

```diff
@@ -17,38 +18,60 @@ const (
 
 	// double represents "double quoted" parts of a BAPS3 message.
 	double
+
+	// bufsize is the number of bytes the Tokeniser will try to read from
+	// its Reader in one go.
+	bufsize int = 4096
 )
 
-// Tokeniser holds the state of a BAPS3 protocol tokeniser.
+// Tokeniser holds the state of a Bifrost protocol tokeniser.
 type Tokeniser struct {
+	// raw is the current back-buffer of bytes to tokenise.
+	raw []byte
+	// rawpos is the current position in `raw`.
+	rawpos int
+	// rawcount is the number of valid bytes in `raw`.
+	rawcount int
+
 	inWord           bool
 	escapeNextChar   bool
 	currentQuoteType quoteType
 	word             *bytes.Buffer
 	words            []string
-	lines            [][]string
+	lineDone         bool
 	err              error
+	reader           io.Reader
 }
 
 // NewTokeniser creates and returns a new, empty Tokeniser.
-func NewTokeniser() *Tokeniser {
+// The Tokeniser will read from the given Reader when Tokenise is called.
+func NewTokeniser(reader io.Reader) *Tokeniser {
+	return NewTokeniserWith(reader, bufsize)
+}
```

Comment:
The double constructor (which I think is unnecessary anyway) also goes away with reading single bytes